src/Avalonia.Visuals/Media/PolylineGeometry.cs

```diff
@@ -15,36 +14,122 @@ namespace Avalonia.Media
     /// </summary>
     public class PolylineGeometry : Geometry
     {
-        private IList<Point> _points;
-        private bool _isFilled;
+        /// <summary>
+        /// Defines the <see cref="Points"/> property.
+        /// </summary>
+        public static readonly DirectProperty<PolylineGeometry, Points> PointsProperty =
+            AvaloniaProperty.RegisterDirect<PolylineGeometry, Points>(nameof(Points), g => g.Points, (g, f) => g.Points = f);
 
-        public PolylineGeometry(IList<Point> points, bool isFilled)
+        /// <summary>
+        /// Defines the <see cref="IsFilled"/> property.
+        /// </summary>
+        public static readonly AvaloniaProperty<bool> IsFilledProperty =
+            AvaloniaProperty.Register<PolylineGeometry, bool>(nameof(IsFilled));
+
+        static PolylineGeometry()
+        {
+            PointsProperty.Changed.Subscribe(onNext: v =>
+            {
+                (v.Sender as PolylineGeometry)?.OnPointsChanged(v.OldValue as Points, v.NewValue as Points);
+            });
+            IsFilledProperty.Changed.AddClassHandler<PolylineGeometry>(x => a => x.NotifyChanged());
```

Comment:
It works because it uses the overload that accepts a `Func<TTarget, Action<AvaloniaPropertyChangedEventArgs>>`. It's the same as writing `x => x.SomeMethod`, but instead, `SomeMethod` is a lambda. Anyway, I didn't know the other overload existed :)